vm/src/vm.rs

```diff
@@ -311,34 +313,29 @@ impl VirtualMachine {
     }
 
     pub fn import(&self, module: &str, from_list: &PyObjectRef, level: usize) -> PyResult {
-        let sys_modules = self.get_attribute(self.sys_module.clone(), "modules")?;
-        sys_modules
```

Comment:
Did you remove the caching mechanism (if name is in `sys_modules`) on purpose?

Reply:
Yes. The first thing the `import` function does is to check for that.